packages/@aws-cdk/aws-eks/test/nodegroup.test.ts

```diff
@@ -230,9 +230,284 @@ describe('node group', () => {
+  test('deploy correctly with defined LaunchTemplate and instanceTypes(GPU) and amiType undefined.', () => {
+    // GIVEN
+    const { stack, vpc } = testFixture();
+
+    // WHEN
+    const cluster = new eks.Cluster(stack, 'Cluster', {
+      vpc,
+      defaultCapacity: 0,
+      version: CLUSTER_VERSION,
+    });
+    const ng = new eks.Nodegroup(stack, 'Nodegroup', {
+      cluster,
+      instanceTypes: [
+        new ec2.InstanceType('p3.large'),
+        new ec2.InstanceType('g3.large'),
+      ],
+      launchTemplateSpec: {
+        id: 'mock',
+      },
+    });
+
+    // THEN
+    expect(ng).not.toHaveProperty('AmiType');
+  });
+  /**
+   * When LaunchTemplate is defined, amiType is undefined and instanceTypes are x86_64 instances,
+   * we should deploy correctly.
+   */
+  test('deploy correctly with defined LaunchTemplate and instanceTypes(x86_64) and amiType undefined.', () => {
+    // GIVEN
+    const { stack, vpc } = testFixture();
+
+    // WHEN
+    const cluster = new eks.Cluster(stack, 'Cluster', {
+      vpc,
+      defaultCapacity: 0,
+      version: CLUSTER_VERSION,
+    });
+    const ng = new eks.Nodegroup(stack, 'Nodegroup', {
+      cluster,
+      instanceTypes: [
+        new ec2.InstanceType('c5.large'),
+        new ec2.InstanceType('m5.large'),
+      ],
+      launchTemplateSpec: {
+        id: 'mock',
+      },
+    });
+
+    // THEN
+    expect(ng).not.toHaveProperty('AmiType');
+  });
+  /**
+   * When LaunchTemplate is defined, amiType is undefined and instanceTypes are ARM_64 instances,
+   * we should deploy correctly.
+   */
+  test('deploy correctly with defined LaunchTemplate and instanceTypes(ARM_64) and amiType undefined.', () => {
+    // GIVEN
+    const { stack, vpc } = testFixture();
+
+    // WHEN
+    const cluster = new eks.Cluster(stack, 'Cluster', {
+      vpc,
+      defaultCapacity: 0,
+      version: CLUSTER_VERSION,
+    });
+    const ng = new eks.Nodegroup(stack, 'Nodegroup', {
+      cluster,
+      instanceTypes: [
+        new ec2.InstanceType('c6g.large'),
+        new ec2.InstanceType('t4g.large'),
+      ],
+      launchTemplateSpec: {
+        id: 'mock',
+      },
+    });
+
+    // THEN
+    expect(ng).not.toHaveProperty('AmiType');
+  });
+
+  /**
+   * BOTTOEROCKET_X86_64 with defined instance types w/o launchTemplateSpec should deploy correctly.
+   */
+  test('BOTTOEROCKET_X86_64 with defined instance types w/o launchTemplateSpec should deploy correctly', () => {
+    // GIVEN
+    const { stack, vpc } = testFixture();
+    const cluster = new eks.Cluster(stack, 'Cluster', {
+      vpc,
+      defaultCapacity: 0,
+      version: CLUSTER_VERSION,
+    });
+    // THEN
+    cluster.addNodegroupCapacity('bottlerocket', {
+      instanceTypes: [new ec2.InstanceType('m5a.xlarge')],
+      amiType: NodegroupAmiType.BOTTLEROCKET_X86_64,
+    });
+
+    // THEN
+    expect(stack).toHaveResourceLike('AWS::EKS::Nodegroup', {
+      AmiType: 'BOTTLEROCKET_x86_64',
+    });
+  });
```

Comment:
A special unit test for the reported issue. FYR @kirintwn

Reply:
m5a.xlarge with BOTTLEROCKET_X86_64 successfully deployed.

![image](https://user-images.githubusercontent.com/278432/145947686-5160d30a-3f71-4be5-8eea-eb6c13ffd2c4.png)

![image](https://user-images.githubusercontent.com/278432/145947728-d5e90377-63c8-4333-8818-c7da31998767.png)